test/menus/connection_menu/events_test.js

```diff
@@ -0,0 +1,116 @@
+require("../../setup");
+
+describe("ConnectionMenu", function() {
+  const helpers = require("./helpers.js");
+  const c = helpers.constants;
+  const ID = "ConnectionMenu-for-methods-test";
+
+  function testSelectionCalled(instance, selector) {
+    var originalSelection = c.ConnectionMenuClass.prototype.selection;
+    var $item = instance.$node.find(selector);
+    var called = false;
+
+    // Fake selection
+    instance.selection = function() {
+      called = true;
+    }
+
+    // Activate event.
+    $item.click();
+
+    // Reset selection() back to original.
+    instance.selection = originalSelection;
+
+    return called;
+  }
+
+  function testActionCalled(instance, selector, func) {
+    var originalAction = c.ConnectionMenuClass.prototype[func];
+    var $item = instance.$node.find(selector);
+    var called = false;
+
+    // Fake selection
+    instance[func] = function() {
+      called = true;
+    }
+
+    // Activate event.
+    $item.click();
+
+    // Reset selection() back to original.
+    instance[func] = originalAction;
+
+    return called;
+  }
+
+
+  describe("Events", function() {
+    var created;
+
+    beforeEach(function() {
+      helpers.setupView(ID);
+      created = helpers.createConnectionMenu(ID);
+    });
+
+    afterEach(function() {
+      helpers.teardownView(ID);
+      created = {};
+    });
+
+
+    /* TEST EVENT: menuselect
+     **/
+    describe("menuselect", function() {
+
+      it("should not call the selection method when menuselect is triggered with action 'none'", function() {
+        var called = testSelectionCalled(created.item, "li[data-action=" + c.TEXT_ACTION_NONE + "]");
+        //expect(called).to.be.false;
```

Comment:
Should this be commented?

Reply:
Well spotted. An earlier effort had me changing code to match that test, which resulted in a positive (ie. expect false) result. However, I back-tracked on that change (for whatever reason I no longer remember) and clearly didn't reverse the mid-development-comment-out effort. 

It's a little pointless since 'none' doesn't do anything inside `selection()` but I'll alter the test to match what is currently does, e.g.
```
      it("should call the selection method when menuselect is triggered with action 'none'", function() {
        var called = testSelectionCalled(created.item, "li[data-action=" + c.TEXT_ACTION_NONE + "]");
        expect(called).to.be.true;
      });
```